tests/parameter/test_parameter_override.py

```diff
@@ -96,3 +119,41 @@ def test_removed_parameter_from_prop_instrument_works(request):
     a.remove_parameter("voltage")
     a.add_parameter("voltage", set_cmd=None, get_cmd=None)
     a.voltage.set(1)
+
+
+def test_remove_parameter_from_class_attr_works(request, caplog):
+    request.addfinalizer(DummyClassAttrInstr.close_all)
+    a = DummyClassAttrInstr("my_instr")
+
+    # removing a parameter that is assigned as an attribute
+    # with a class level type works
+    assert hasattr(a, "current")
+    a.remove_parameter("current")
+    assert not hasattr(a, "current")
+
+    # when we remove a parameter with an attr that shadows a class attr
+    # we get the class attr after the removal
```

Comment:
This behavior might be unexpected. If we keep it this way, we should document it.

Reply:
@einsmein This is exactly the same way as instance attributes and class attributes already work in python so I don't think this is surprising as such. And if it is surprising then this is a thing that should be documented in python the language not here. 

Furthermore, we already raise a warning when a parameter overrides an attribute so this is not really a situation that anyone should get in unless their code is already generating warnings. 

```python
In [11]: class Foo():
    ...:     a: str = "classattr"
    ...:     def __init__(self):
    ...:         self.a = 1
    ...:

In [12]: foo = Foo()

In [13]: foo.a
Out[13]: 1

In [14]: delattr(foo, 'a')

In [15]: foo.a
Out[15]: 'classattr'
```